### PR TITLE
Added JSON support for NullDecimal

### PR DESCRIPTION
**Pull request by MOZGIII**

### CODE REVIEW

File: decimal.go [hunk 1]
```diff
@@ -928,8 +928,8 @@ func unquoteIfQuoted(value interface{}) (string, error) {
 	return string(bytes), nil
 }
 
-// NullDecimal represents a fixed-point decimal. It is immutable.
-// number = value * 10 ^ exp
+// NullDecimal represents a nullable decimal with compatibility for
+// scaning null values from the database.
```
Comment (victorquinn) [inline decimal.go]: Minor spelling error, "scaning" should be "scanning" -- otherwise this comment update is much needed 👍

Comment (MOZGIII) [reply decimal.go]: Oops, I'll fix that

**Comment by MOZGIII** [comment]
Added tests

**Comment by MOZGIII** [comment]
I instantly found a few issues in test, lol

**Comment by MOZGIII** [comment]
LGTM?

**Comment by victorquinn** [comment]
Sorry, LGTM = Looks Good To Me

**Comment by MOZGIII** [comment]
Wow, I though it's let's get this merged. :/

**Pull request merged**